### ISSUE TITLE
Support for Piped (Invidious alternative)

### ISSUE THREAD

**Issue by Victor239** [FreeTubeApp/FreeTube#1527]
**Is your feature request related to a usage problem (not a bug)? Please describe.**
[Piped](https://github.com/TeamPiped/Piped) is a FOSS Invidious alternative. It would be great to be able to use Piped instances instead of Invidious ones for playing videos.

**Describe the solution you'd like to see implemented**
Instead of selecting "Invidious Instance", ability to select either a Invidious or a Piped instance.

**Describe alternatives you've considered**
N/A

**Comment by PrestonN** [comment]
I do not believe they provide an open API. Even if they did, they would need to structure the data to be exactly like Invidious. If they didn't then we'd have to do a tremendous amount of work to implement it to the point where the same amount of effort could be used to implement something like PeerTube instead.

Because of these reasons, it seems unlikely that it'll be supported.

**Comment by peepopoggers** [comment]
@PrestonN They have an API but it is [undocumented as of now](https://github.com/TeamPiped/Piped/issues/395)
It is better to keep this issue open so we can discuss changes made by the Piped team.

**Comment by PrestonN** [comment]
My argument still stands where even if they have an API, there's way too much work that would need to be done to support it if it is going to be structured differently to the Invidious API. The amount of effort it would take to support Piped would be similar to if we supported an entirely new platform such as PeerTube. If we're going to be making that effort, I would much prefer that time be spent on actually adding PeerTube support rather than adding _another_ way of accessing YouTube. If there comes a time where Piped _did_ structure their API to be the same as Invidious, then we wouldn't have to do anything towards it anyway.

Either way, there's nothing on our end that's going to be done about it, and there isn't a reason that the issue should stay open because of it.

**Comment by peepopoggers** [comment]
Okay.
If they add documentation, and if I have time, I will be the one to add support.

**Comment by peepopoggers** [comment]
update: https://piped-docs.kavin.rocks/docs/api-documentatoin/

**Comment by kubo6472** [comment]
fixed link: https://piped-docs.kavin.rocks/docs/api-documentation/

**Comment by ChunkyProgrammer** [comment]
Piped now has API documentation: https://github.com/TeamPiped/OpenAPI

Reopening this issue as I think this is more possible now that we're decoupling a lot of the api code from the views/components (we want to parse all api data so that they follow the same format => we can make the piped response be the same as invidious/local api)

I don't think this will be added for a while and I'm not sure if anyone on the dev team would want to work on this but if someone else created a PR for it, I don't see a reason why it wouldn't be merged once #3048 is finished.

**Comment by Cyanide-1** [comment]
honestly i just found this open issue, i would personally love to see this get merged at some point, adding more API's would allow for greater flexibility in the usability of freetube and allow access to more servers (aka the ones from piped), i find that a lot of invidious servers for instance are really slow and not really usable, there are a couple of fast ones but few and far between, but there some good piped servers i use on mobile (i use libretube on mobile which uses piped as a backend for api)

i think this would be a great boon to freetube, im fully in support of this getting merged and would love to see it personally

**Comment by noraj** [comment]
At least, we could have Piped subscription format import (JSON)?

**Comment by FXlife** [comment]
Hi. I just wonder if this issue or better feature request is still in consideration. Invidious instances are almost inexistant currently (in 2025). Switching to Piped instances or APIs are in my humble opinion the most viable option to let FreeTube survive.

I use FreeTube for various personal reasons but this would help avoid cumbersomesearches to find a workable and suitable Invidious instance.
